table/tables/cache.go

```diff
@@ -72,31 +84,48 @@ func (c *cachedTable) TryReadFromCache(ts uint64) kv.MemBuffer {
 	}
 	data := tmp.(*cacheData)
 	if ts >= data.Start && ts < data.Lease {
+		leaseTime := oracle.GetTimeFromTS(data.Lease)
+		nowTime := oracle.GetTimeFromTS(ts)
+		distance := leaseTime.Sub(nowTime)
+		// TODO make this configurable in the following PRs
+		if distance <= (1 * time.Second) {
```

Comment:
It could be a negnative number, which means the lease is already gone and you shouldn't renew it.